### ISSUE TITLE
Catch typo with :offset expansion

### ISSUE THREAD

**Issue by dimo414** [koalaman/shellcheck#2048]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
echo "${1:missing arg}"
```

#### Here's what shellcheck currently says:
```
No issues detected!
```
#### Here's what I wanted or expected to see:

I found a typo where I'd intended to write `${1:?missing arg}` but left off the `?`. This erroneously turns the expansion into an [offset expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html), which (AFAICT) silently ignores non-numeric values.

It would be great if ShellCheck could flag usages of `${var:...}` where `...` isn't a numeral (or a variable).
